docs/intro/fastapi_gelai_searchbot.rst

```diff
@@ -0,0 +1,1556 @@
+
+.. image::
+    /docs/tutorials/placeholder.png
+    :alt: Placeholder
+    :width: 100%
+
+1. Initialize the project
+=========================
+
+We're going to start by installing `uv <https://docs.astral.sh/uv/>`_ - a Python
+package manager that's going to simplify environment management for us. You can
+follow their `installation instructions
+<https://docs.astral.sh/uv/getting-started/installation/>`_ or simply run:
+
+.. code-block:: bash
+
+    $ curl -LsSf https://astral.sh/uv/install.sh | sh
+
+Once that is done, we can use uv to create scaffolding for our project following
+the `documentation <https://docs.astral.sh/uv/guides/projects/>`_:
+
+.. code-block:: bash
+
+    $ uv init searchbot \
+      && cd searchbot
+
+For now, we know we're going to need Gel and FastAPI, so let's add those
+following uv's instructions on `managing dependencies
+<https://docs.astral.sh/uv/concepts/projects/dependencies/#optional-dependencies>`_,
+as well as FastAPI's `installation docs
+<https://fastapi.tiangolo.com/#installation>`_. Running ``uv sync`` after that
+will create our virtual environment in a ``.venv`` directory and ensure it's
+ready. As a last step, we'll activate the environment and get started.
+
+.. code-block:: bash
+
+    $ uv add "fastapi[standard]" \
+      && uv add gel \
+      && uv sync \
+      && source .venv/bin/activate
+
+.. note::
+
+    Make sure to source the environment every time you open a new terminal
+    session before running ``python``, ``gel`` or ``fastapi``-related commands.
+
+
+2. Get started with FastAPI
+===========================
+
+At this stage we need to follow FastAPI's `tutorial
+<https://fastapi.tiangolo.com/tutorial/>`_ to create the foundation of our app.
+
+We're going to make a super simple web API with one endpoint that takes in a
+user query as an input and echoes it as an output. First, let's make a directory
+called ``app`` in our project root, and put an empty ``__init__.py`` there.
+
+.. code-block:: bash
+
+   $ mkdir app && touch app/__init__.py
+
+Now let's create a file called ``main.py`` inside the ``app`` directory and put
+the "Hello World" example in it:
+
+.. code-block:: python
+    :caption: app/main.py
+
+    from fastapi import FastAPI
+
+    app = FastAPI()
+
+
+    @app.get("/")
+    async def root():
+        return {"message": "Hello World"}
+
+To start the server, we'll run:
+
+.. code-block:: bash
+
+    $ fastapi dev app/main.py
+
+Once the server gets up and running, we can make sure it works using FastAPI's
+built-in UI at <http://127.0.0.1:8000/docs>_, or manually by using ``curl``:
+
+.. code-block:: bash
+
+    $ curl -X 'GET' \
+      'http://127.0.0.1:8000/' \
+      -H 'accept: application/json'
+
+    {"message":"Hello World"}
+
+Now, in order to create the endpoint we set out to create, we need to pass our
+query as a parameter to it. We'd prefer to have it in the body of the request
+since user messages can get pretty long.
```

Comment:
```suggestion
Now, to create the search endpoint we mentioned earlier, we need to pass our
query as a parameter to it. We'd prefer to have it in the request's body
since user messages can be long.
```